integration/spark2/src/main/scala/org/apache/spark/sql/CarbonEnv.scala

```diff
@@ -184,6 +184,9 @@ object CarbonEnv {
       .addListener(classOf[LoadTablePostExecutionEvent], new MergeIndexEventListener)
       .addListener(classOf[AlterTableCompactionPostEvent], new MergeIndexEventListener)
       .addListener(classOf[AlterTableMergeIndexEvent], new MergeIndexEventListener)
+      .addListener(classOf[LoadTablePreStatusUpdateEvent], new MergeBloomIndexEventListener)
```

Comment:
After adding this line, the segment will not be visible to user until the mergeBloomIndex procedure finish, right?

Reply:
Yes. This is what this PR wants to do.

Reply:
yeah, that is supposed to be.

Besides, I think it's framework's responsibility to include this procedure in the data-loading transaction. So the best practice is to optimize the framework's behavior.

@jackylk How do you think about this. Should we do this in this PR or later?

Reply:
@jackylk @ravipesala @KanakaKumar Please check and review